src/components/ComputeModal.js

```diff
@@ -328,6 +330,24 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
     onSuccess()
   })
 
+  const getMainMachineTypeByNumCpus = numCpus => _.find({ cpu: numCpus }, validMachineTypes)?.name || mainMachineType
+
+  const getMainMachineTypeByMemory = (numCpus, memory) => _.find({ cpu: numCpus, memory }, validMachineTypes)?.name || mainMachineType
+
+  const getValidCpuGpuConfig = machineType => {
+    const { cpu: currentNumCpus, memory: currentMemory } = findMachineType(machineType)
+    const validGpuOptions = getValidGpuOptions(currentNumCpus, currentMemory, computeConfig.computeZone)
+    const validGpuNames = _.flow(_.map('name'), _.uniq, _.sortBy('price'))(validGpuOptions)
+    const validGpuName = _.includes(displayNameForGpuType(computeConfig.gpuType), validGpuNames) ?
+      displayNameForGpuType(computeConfig.gpuType) :
+      _.head(validGpuNames)
+    const validGpuType = _.find({ name: validGpuName }, validGpuOptions)?.type
+    const validNumGpusOptions = _.flow(_.filter({ name: validGpuName }), _.map('numGpus'))(validGpuOptions)
+    const validNumGpus = _.includes(computeConfig.numGpus, validNumGpusOptions) ? computeConfig.numGpus : _.head(validNumGpusOptions)
+
+    return { currentNumCpus, currentMemory, validGpuName, validGpuNames, validGpuType, validGpuOptions, validNumGpus, validNumGpusOptions }
+  }
+
```

Comment:
These now need to be computed in multiple places so I factored out the code that used to be on top of `renderComputeProfileSection` below and made it a helper function.